File: Bit Manipulation/Minimum-Updates-to-Make-Bitwise-OR-Equal-to-Target.cpp
int solve(int first, int second, int target) {
  int updates = 0;

  for (int bit = 0; bit < 30; ++bit) {
    const int first_bit = first & (1 << bit),
      second_bit = second & (1 << bit),
      target_bit = target & (1 << bit);
    
    if (target_bit) {
      if (!first_bit && !second_bit) {
        ++updates;
      }
    } else {
      if (first_bit && second_bit) {
        updates += 2;
      } else if (first_bit || second_bit) {
        ++updates;
      }
    }
  }

  return updates;
}

File: Bit Manipulation/Minimum-Updates-to-Make-Bitwise-OR-Equal-to-Target.py
class Solution:
  def solve(self, first, second, target):
    def cost(first_bit, second_bit, target_bit):
      if first_bit and second_bit and not target_bit:
        return 2
      if ((first_bit or second_bit) and not target_bit) or \
          (not first_bit and not second_bit and target_bit):
        return 1
      return 0

    return sum(cost(first & (1 << bit), second & (1 << bit), target & (1 << bit)) for bit in range(30))
